src/aws/ec2-wrapper.ts

```diff
@@ -80,22 +64,22 @@ export class EC2Wrapper {
     );
 
     if (subnets.length === 0) {
-      throw new Error("Invalid subnet name, it does not exist");
+      throw new Error(`Subnets with vpc id '${vpcId}', tag key '${tagKey}' and tag values '${tagValues}' do not exist`);
     }
 
-    const missingSubnetNames = subnetNames.filter((subnetName) => {
+    const missingSubnetValues = tagValues.filter((tagValue) => {
       // collect subnets by name
       const subnetsByName = subnets.filter((subnet) => {
-        const nameTag = subnet.Tags.find((tag) => tag.Key === "Name");
-        return wildcardMatches(subnetName, nameTag.Value);
+        const nameTag = subnet.Tags.find((tag) => tag.Key === tagKey);
```

Comment:
Nitpick: A util function like `tagsToMap` might be helpful or maybe a function like `getValueFromTags(tags, tagKey)`